### PR TITLE
[WIP] IR279 - Add convenient initializers for blob_t

### PR DESCRIPTION
**Pull request by Warchant**
## What is this pull request?
- [x] add `blob_t<N>::from_string(std::string s);` which creates blob_t of size N from strings
- [x] add `blob_t<N>::from_hexstring(std::string s);` which creates blob_t of size N from string encoded in hex (like 0x1234567890abcdef)
- [x] add `blob_t<N>::from_string_var(std::string s);` which creates blob_t of size N from strings of variable size. Example `blob_t<32>::from_string_var("1"); // result is 1000...00 (32 bytes)`
- [x] add tests for each of method and constructor
- [x] fixed few bugs in tests related to "bad format" for pubkey and signature
- [x] refactored the code, to get rid of `std::copy`

Anything else?

### CODE REVIEW

**Comment by Warchant** [comment]
TODO: please, review tests once more. Some of them are broken after my update.
The reason is that public keys are created from strings of incorrect lengths, thus tests fail with exceltion `std::invalid_argument`

File: libs/crypto/ed25519_impl.cpp [hunk 1]
```diff
@@ -16,12 +16,19 @@
  */
 
 #include <ed25519.h>
```
Comment (takemiyamakoto) [inline libs/crypto/ed25519_impl.cpp]: I don't really like that this file is called ed25519_impl. We don't have hash_impl, etc. It feels somewhat inconsistent to me.

File: irohad/ametsuchi/impl/postgres_wsv_query.cpp [hunk 2]
```diff
@@ -55,8 +55,7 @@ namespace iroha {
       row.at("account_id") >> account.account_id;
       row.at("domain_id") >> account.domain_name;
       pqxx::binarystring master_key(row.at("master_key"));
-      std::copy(master_key.begin(), master_key.end(),
-                account.master_key.begin());
+      account.master_key = std::string{master_key.begin(), master_key.end()};
```
Comment (lebdron) [inline irohad/ametsuchi/impl/postgres_wsv_query.cpp]: Use `pqxx::binarystring::str()`

**Comment by neewy** [comment]
We have agreed to use optional type + monads for error handling instead of catching exceptions

**Comment by neewy** [comment]
please, use monads the same way as in https://github.com/hyperledger/iroha/pull/552 @Warchant

**Comment by neewy** [comment]
Assigned "candidate for closing" since https://github.com/hyperledger/iroha/pull/552 was accepted

**Comment by neewy** [comment]
@Warchant is requested to review https://github.com/hyperledger/iroha/pull/552

**Comment by neewy** [comment]
@Warchant please review this PR

**Pull request closed without merging**